File: packages/models/maze.py
from pydantic import BaseModel


class PromptTokens(BaseModel):
    model_config = {"extra": "forbid"}
    
    subject_token: str
    action_token: str
    environment_token: str
    
    def to_sentence(self) -> str:
        return f"{self.subject_token} {self.action_token} {self.environment_token}"


EXAMPLE_TOKENS = {
    "subject_token": "cat",
    "action_token": "jumps",
    "environment_token": "rooftop"
}


if __name__ == "__main__":
    print(PromptTokens(**EXAMPLE_TOKENS).to_sentence()) 

File: packages/models/maze.ts
import { z } from "zod";

export const promptTokensSchema = z.object({
  subject_token: z.string(),
  action_token: z.string(),
  environment_token: z.string(),
}).strict();

export type PromptTokens = z.infer<typeof promptTokensSchema>;

export function toSentence(tokens: PromptTokens): string {
  return `${tokens.subject_token} ${tokens.action_token} ${tokens.environment_token}`;
}

export const EXAMPLE_TOKENS = {
  subject_token: "cat",
  action_token: "jumps",
  environment_token: "rooftop",
}; 